core/Segment.php

```diff
@@ -180,17 +180,30 @@ public function getSegmentExpression()
         return $this->segmentExpression;
     }
 
+    /**
+     * @throws Exception
+     */
     private function getAvailableSegments()
     {
+        // start cache
+        $cache = PiwikCache::getTransientCache();
+
+        //covert cache id
+        $cacheId = 'API.getSegmentsMetadata'.SettingsPiwik::getPiwikInstanceId() . implode(",", $this->idSites);
```

Comment:
Even though it might be a very rare race condition and shouldn't even happen with transient cache, I would suggest to add some additional separators to the cache key.
Assuming two instances `matomo` and `matomo2` and each having multiple sites, the cache key for `matomo` with site id `21` and `matomo2` with site id `1`, would currently be the same.
```suggestion
        $cacheId = 'API.getSegmentsMetadata.'.SettingsPiwik::getPiwikInstanceId() . '.' . implode(",", $this->idSites);
```